src/main/java/seedu/address/logic/commands/ReassignCommand.java

```diff
@@ -0,0 +1,147 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ROLE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_SHIFT_NEW;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_SHIFT_OLD;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_WORKER;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_SHIFTS;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_WORKERS;
+
+import java.util.List;
+import java.util.Set;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.assignment.Assignment;
+import seedu.address.model.assignment.exceptions.AssignmentNotFoundException;
+import seedu.address.model.shift.Shift;
+import seedu.address.model.tag.Role;
+import seedu.address.model.worker.Unavailability;
+import seedu.address.model.worker.Worker;
+
+public class ReassignCommand extends Command {
+    public static final String COMMAND_WORD = "reassign";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Reassigns a worker to a new shift in the "
+            + "McScheduler by the index numbers used in the last worker and shift listings. "
+            + "\nParameters: "
+            + PREFIX_WORKER + "WORKER_INDEX (must be a positive integer) "
```

Comment:
Would it be possible to allow the editing of the worker as well?

Reply:
Good point, will make the change!